benchmarks_test.go

```diff
@@ -31,16 +27,31 @@ type fetchFunc func(b *testing.B, bs *bitswap.Bitswap, ks []cid.Cid)
 
 type distFunc func(b *testing.B, provs []testinstance.Instance, blocks []blocks.Block)
 
-type runStats struct {
-	DupsRcvd uint64
-	BlksRcvd uint64
-	MsgSent  uint64
-	MsgRecd  uint64
-	Time     time.Duration
-	Name     string
+type stats struct {
+	messagesRecvd, messagesSent, dupBlocksReceived, blocksReceived uint64
 }
 
-var benchmarkLog []runStats
+func (s *stats) record(fetchers ...testinstance.Instance) error {
+	for _, fetcher := range fetchers {
+		netstats := fetcher.Adapter.Stats()
+		exstats, err := fetcher.Exchange.Stat()
+		if err != nil {
+			return err
+		}
+		s.messagesRecvd += netstats.MessagesRecvd
+		s.messagesSent += netstats.MessagesSent
+		s.dupBlocksReceived += exstats.DupBlksReceived
+		s.blocksReceived += exstats.BlocksReceived
+	}
+	return nil
+}
+
+func (s *stats) report(b *testing.B) {
+	b.ReportMetric(float64(s.messagesRecvd)/float64(b.N), "msg-recvd/op")
+	b.ReportMetric(float64(s.messagesSent)/float64(b.N), "msg-sent/op")
+	b.ReportMetric(float64(s.blocksReceived)/float64(b.N), "blocks-recvd/op")
+	b.ReportMetric(float64(s.dupBlocksReceived)/float64(b.N), "dups-recvd/op")
```

Comment:
Nice!

(Came here to say: "have you seen the new `B.ReportMetric` feature?" and I'm beaten to it.  I just recently learned of it, and I think this will be the first time I see it used :))